pkg/planner/funcdep/fd_graph.go

```diff
@@ -1224,3 +1224,51 @@ func (s *FDSet) IsHashCodeRegistered(hashCode string) (int, bool) {
 	}
 	return -1, false
 }
+
+// FindCommonEquivClasses find the common equivalences between multi fds.
+func FindCommonEquivClasses(fdSets []*FDSet) []intset.FastIntSet {
+	if len(fdSets) == 0 {
+		return nil
+	}
+	var result []intset.FastIntSet
+	for _, edge := range fdSets[0].fdEdges {
+		if edge.equiv {
+			// clone one for later mutable intersection.
+			set := intset.NewFastIntSet()
+			set.CopyFrom(edge.from)
+			result = append(result, set)
+		}
+	}
+	// iterate intersection with later all fdSets.
+	for i := 1; i < len(fdSets); i++ {
+		var newResult []intset.FastIntSet
+		currFDSet := fdSets[i]
+		// iterate all equivalence classes in the current fdSet.
+		for _, equivClass := range result {
+			// iterate all equivalence FDs in the current fdSet.
+			for _, edge := range currFDSet.fdEdges {
+				if !edge.equiv {
+					continue
+				}
+				// find the intersection between the current equivalence class and the current equivalence FD.
+				intersection := intset.NewFastIntSet()
+				intersection.CopyFrom(equivClass)
+				intersection.IntersectionWith(edge.from)
+				// * if the intersection has changed and not empty. (become a subset)
+				// * if the intersection is still the same (it means two equivalence classes are the same)
+				if !intersection.IsEmpty() {
```

Comment:
I think we can remove the equiv class who only contains 1 element

Reply:
make sense